src/main/java/com/emc/mongoose/storage/driver/coop/netty/http/s3/S3StorageDriver.java

```diff
@@ -157,9 +173,57 @@ public S3StorageDriver(
 		if (versioning) {
 			Loggers.MSG.info("Versioning is enabled. Make sure that if you use input items list it has version ids");
 		}
+		var authVersionTemp = storageConfig.intVal("auth-version");
+		if ((authVersionTemp != 2) && (authVersionTemp != 4)) {
+			Loggers.MSG.warn("S3 auth version is not 2 or 4. Resetting to 2.");
```

Comment:
In context of using S3 driver, would it be a good practice to reset the version to default value in case it's not supported? In other areas, I'd prefer to throw an exception prior to the start up in case configuration is not valid.